File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/ApiErrorDTO.java
package com.thanh0x.coursedeal.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * Standardized error response for the API.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ApiErrorDTO {
    private Date timestamp;
    private int status;
    private String error;
    private String message;
    private String path;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/ApiErrorDTO.kt
package com.thanh0x.coursedeal.dto

import java.util.Date

/**
 * Standardized error response for the API.
 */
data class ApiErrorDTO(
    val timestamp: Date? = null,
    val status: Int = 0,
    val error: String? = null,
    val message: String? = null,
    val path: String? = null
)
